crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -687,43 +833,55 @@ impl Blockchain {
             fulfillments.push(Fulfillment {
                 quoted_order: self.quote(order),
                 execution: execution.clone(),
-                interactions: vec![
-                    Interaction {
-                        address: sell_token.address(),
-                        calldata: match solution.calldata {
-                            super::Calldata::Valid { additional_bytes } => transfer_interaction
-                                .into_iter()
-                                .chain(std::iter::repeat(0xab).take(additional_bytes))
-                                .collect(),
-                            super::Calldata::Invalid => vec![1, 2, 3, 4, 5],
-                        },
-                        inputs: Default::default(),
-                        outputs: Default::default(),
-                        internalize: false,
-                    },
-                    Interaction {
-                        address: pair.contract.address(),
-                        calldata: match solution.calldata {
-                            super::Calldata::Valid { .. } => swap_interaction,
-                            super::Calldata::Invalid => {
-                                vec![10, 11, 12, 13, 14, 15, 63, 78]
-                            }
-                        },
-                        inputs: vec![eth::Asset {
-                            token: sell_token.address().into(),
-                            // Surplus fees stay in the contract.
-                            amount: (execution.sell - order.surplus_fee()).into(),
-                        }],
-                        outputs: vec![eth::Asset {
-                            token: buy_token.address().into(),
-                            amount: execution.buy.into(),
-                        }],
-                        internalize: order.internalize,
-                    },
-                ],
+                interactions: (!is_jit_order)
+                    .then(|| {
```

Comment:
Since jit orders don't have interactions, we can probably slightly refactor the structs and remove `interactions` field from the Trade::Jit completely.

Reply:
According to @MartinquaXD , JIT orders could have interactions too.